src/XrdPfc/XrdPfcFile.cc

```diff
@@ -1428,20 +1429,22 @@ void File::ProcessBlockResponse(Block *b, int res)
                }
             }
          }
-
-         // If failed with no subscribers -- delete the block and exit.
-         if (b->m_refcnt == 0 && (res < 0 || m_in_shutdown))
-         {
-            free_block(b);
-            m_state_cond.UnLock();
-            return;
-         }
-         m_prefetch_bytes += b->get_size();
       }
       else
       {
          TRACEF(Error, tpfx << "io " << b->get_io() << " not found in IoSet.");
       }
+
+      // If nothing is subscribed to this block -- or if the originating I/O object
+      // is gone -- and the block read failed/incomplete or the file is in shutdown,
+      // then we do not want to re-issue the read later.  Immediately free and return.
+      if ((b->m_refcnt == 0 || !found_io) && (res != b->get_size() || m_in_shutdown))
```

Comment:
please change `res != b->get_size()` to `res < 0` (as it was in the original conditional). the read size mismatch case is handled at the top, at this stage `res < 0` is the only thing it matters.

Reply:
Addressed in the latest push.